src/v/coproc/script_context.cc

```diff
@@ -174,14 +175,13 @@ ss::future<> script_context::process_one_reply(process_batch_reply::data e) {
         return ss::now();
     }
     auto ntp_ctx = found->second;
-    return write_materialized(materialized_ntp, std::move(*e.reader))
-      .then([this, ntp_ctx](write_response wr) {
-          if (wr == write_response::crc_failure) {
+    model::term_id highest_term = ntp_ctx->log.offsets().dirty_offset_term;
```

Comment:
I didn't choose that method because I was concerned how new segments would work if I always chose the same term id. Technically there never would be more then 1 segment if I always set every records term id to 1 right?

Reply:
@dotnwat good idea! Talked with Michal offline about this, turns out i incorrectly assumed the term id was related to the number of segments in storage, this is not true. Term id actually should always be 1 in materialized logs since there will never be a raft promotion.